File: Maximum Pair Sum 1/main.java
import java.util.*;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        long[] arr = new long[n];

        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextLong();
        }

        long first = Long.MIN_VALUE;
        long second = Long.MIN_VALUE;

        for (long x : arr) {
            if (x > first) {
                second = first;
                first = x;
            } else if (x > second) {
                second = x;
            }
        }

        System.out.println(first + second);
    }
}

File: Maximum Pair Sum 1/main.cpp


#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;

    vector<long long> arr(n);
    for (int i = 0; i < n; i++) cin >> arr[i];

    long long first = LLONG_MIN, second = LLONG_MIN;

    for (auto x : arr) {
        if (x > first) {
            second = first;
            first = x;
        } else if (x > second) {
            second = x;
        }
    }

    cout << first + second;
    return 0;
}
